src/Gaffer/PlugAlgo.cpp

```diff
@@ -532,6 +533,66 @@ IECore::DataPtr getValueAsData( const ValuePlug *plug )
 
 }
 
+IECore::DataPtr getArrayAsVectorData( const ArrayPlug *plug )
+{
+	size_t size = plug->children().size();
+	if ( !size )
+	{
+		return nullptr;
+	}
+	auto *childPlug = plug->getChild( 0 );
+	int i = 0;
+	switch(static_cast<Gaffer::TypeId>(childPlug->typeId()))
+	{
+		case FloatPlugTypeId :
+		{
+			FloatVectorDataPtr data = new FloatVectorData();
+			data->writable().resize( size );
+			for( Plug::InputIterator it( plug ); !it.done(); ++it, ++i )
+			{
+				data->writable()[i] = static_cast<const FloatPlug *>( it->get() )->getValue();
+			}
```

Comment:
May not be too important in this case, but in general in Gaffer: there's a non-zero cost to writable() ( it checks a refCount to implement copy-on-write ), so we generally prefer to do: `auto &writable = data->writable()` outside the loop, and use the reference inside the loop, instead of repeating the call to writable. I haven't profiled this recently ... probably actually not a real concern here where the getValue call is a lot more expensive - but it does make a difference in really extreme cases ( like doing an arithmetic operation on every pixel in a tile ), and we've gotten into the habit of doing it everywhere. Probably not important here, but figured I'd mention it.